<launch>

    <arg name="how_many_robot" value="3"/>
    <param name="explore-lite" value="true"/>
    <arg name="cluster_radius" value="2.7"/>
    <param name="cluster_radius" value="$(arg cluster_radius)"/>
    <arg name="world" value="E40-2"/>
    <param name="world" value="$(arg world)"/>

    <!-- world -->
    <include file="$(find test_unknown_rendezvous)/launch/world_sdf.launch">
    </include>

    <!-- robot 1 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="1"/>
        <arg name="pose_x"       value="3"/>
        <arg name="pose_y"       value="13"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="true"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="false"/>
        <arg name="blobbing"     value="false"/>
    </include>

    <!-- robot 2 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="2"/>
        <arg name="pose_x"       value="20"/>
        <arg name="pose_y"       value="10"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="false"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="false"/>
        <arg name="blobbing"     value="false"/>
    </include>
    
    <!-- robot 3 -->
    <include file="$(find test_unknown_rendezvous)/launch/one_robot.launch">
        <arg name="robot_number" value="3"/>
        <arg name="pose_x"       value="4"/>
        <arg name="pose_y"       value="33"/>
        <arg name="pose_z"       value="0"/>
        <arg name="slam"         value="false"/>
        <arg name="formation"    value="false"/>
        <arg name="exploration"  value="false"/>
        <arg name="blobbing"     value="false"/>
    </include>

    <group ns="robot1">
        <node pkg="explore_lite" type="explore" respawn="false" name="explore" output="log">
            <param name="robot_base_frame"      value="/robot1/base_link"/>
            <param name="costmap_topic"         value="/robot1/map"/>
            <param name="costmap_updates_topic" value="/robot1/map_updates"/>
            <param name="visualize"             value="true"/>
            <param name="planner_frequency"     value="1"/>
            <param name="progress_timeout"      value="30"/>
            <param name="potential_scale"       value="3.0"/>
            <param name="orientation_scale"     value="0.0"/>
            <param name="gain_scale"            value="1.0"/>
            <param name="transform_tolerance"   value="0.3"/>
            <param name="min_frontier_size"     value="0.6"/>
        </node>
    </group>


    <!-- all_positions -->
    <node 
        name   = "pos_aggregator"
        pkg    = "test_unknown_rendezvous"
        type   = "pos_aggregator.py"
        args   = "$(arg how_many_robot)"
        output = "screen"
    />
    
    <!-- cluster -->
    <node 
        name   = "cluster"
        pkg    = "test_unknown_rendezvous"
        type   = "cluster_controller.py"
        args   = "$(arg cluster_radius) $(arg how_many_robot)"
        output = "screen"
    />

    <!-- RVIZ -->
    <include file="$(find test_unknown_rendezvous)/launch/rviz.launch">
        <arg name="rviz_file" value="test_8"/>
    </include> 
</launch>